keylime/db/keylime_db.py

```diff
@@ -26,45 +27,66 @@ def make_engine(self, service):
 
-        database_url = config.get(service, 'database_url')
-        if database_url:
-            engine = create_engine(database_url)
-            return engine
-
-        # TODO(kaifeng) Remove following code as well as related configuration
-        # options when the deprecation period is reached.
-        logger.warning('database_url is not set, using deprecated database '
-                       'configuration options')
-        drivername = config.get(service, 'drivername')
-        if drivername == 'sqlite':
-            database = "%s/%s" % (config.WORK_DIR,
-                                  config.get(service, 'database'))
-            # Create the path to where the sqlite database will be store with a perm umask of 077
-            os.umask(0o077)
-            kl_dir = os.path.dirname(os.path.abspath(database))
-            if not os.path.exists(kl_dir):
-                os.makedirs(kl_dir, 0o700)
-
-            url = URL(
-                drivername=drivername,
-                username=None,
-                password=None,
-                host=None,
-                database=(database)
-            )
-            engine = create_engine(url, connect_args={'check_same_thread': False},)
-        else:
-            url = URL(
-                drivername=drivername,
-                username=config.get(service, 'username'),
-                password=config.get(service, 'password'),
-                host=config.get(service, 'host'),
-                database=config.get(service, 'database')
-            )
-            engine = create_engine(url)
+        try :
+            p_sz_m_ovfl = config.get(service, 'database_pool_sz_ovfl')
+            p_sz, m_ovfl = p_sz_m_ovfl.split(',')
+        except NoOptionError :
+            p_sz = 5
+            m_ovfl = 10
 
-        return engine
+        engine_args = {}
+
+        url = config.get(service, 'database_url')
+        if url:
+            logger.info('database_url is set, using it for establishing database connection')
+            engine_args['pool_size'] = int(p_sz)
+            engine_args['max_overflow'] = int(m_ovfl)
+
+        else :
+            logger.info('database_url is not set, using multi-parameter database configuration options')
+
+            # This code shall be removed once we fully deprecate the old format
+            try :
+                drivername = config.get(service, 'drivername')
+                database = config.get(service, 'database')
+                logger.warning('Deprecation reminder: please add the suffix "database_" to all database-related parameters on your keylime.conf.')
+                p_n_prefix = ''
+            except NoOptionError :
+                drivername = config.get(service, 'database_drivername')
+                p_n_prefix = "database_"
+                database = config.get(service, p_n_prefix + 'name')
 
+            if drivername == 'sqlite':
+                database = "%s/%s" % (config.WORK_DIR, database)
```

Comment:
Maybe this new assignment should be to a new variable named something like `database_file` instead of overwriting the original `database` var. Or the other var should be called `database_name` instead of just `database`.